addons/binding/org.openhab.binding.windcentrale/src/main/java/org/openhab/binding/windcentrale/handler/WindCentraleHandler.java

```diff
@@ -0,0 +1,156 @@
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.windcentrale.handler;
+
+import static org.openhab.binding.windcentrale.WindCentraleBindingConstants.*;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.io.IOUtils;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/* The {@link WindCentraleHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Marcel Verpaalen - Initial contribution
+ */
+public class WindCentraleHandler extends BaseThingHandler {
+
+    private int millId = 131;
+    BigDecimal wd = BigDecimal.ONE;
+    private Logger logger = LoggerFactory.getLogger(WindCentraleHandler.class);
+    private ScheduledFuture<?> pollingJob;
+
+    public WindCentraleHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command == RefreshType.REFRESH) {
+            logger.debug("Refreshing {}", channelUID);
+            updateData();
+        } else {
+            logger.warn("This binding is a read-only binding and cannot handle commands");
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing WindCentrale handler '{}'", getThing().getUID());
+
+        Object param;
+
+        param = getConfig().get(PROPERTY_MILL_ID);
+        if (param instanceof BigDecimal && param != null) {
+            millId = ((BigDecimal) param).intValue();
+        } else {
+            millId = 1;
+        }
+
+        param = getConfig().get(PROPERTY_QTY_WINDDELEN);
+        if (param instanceof BigDecimal && param != null) {
+            wd = (BigDecimal) param;
+        } else {
+            wd = BigDecimal.ONE;
+        }
+
+        int pollingPeriod = 30;
+        param = getConfig().get(PROPERTY_REFRESH_INTERVAL);
+        if (param instanceof BigDecimal && param != null) {
```

Comment:
When param is `null` it is no instanceof BigDecimal, so the null check is not necessary.